pkg/iris/scan.go

```diff
@@ -67,64 +76,141 @@ func (s *scanner) Scan(config *rest.Config, workspaceId string, configurationNam
+		return err
 	}
-	kubernetesAPI, err := kubernetes.NewAPI(config)
+
+	kubernetesAPI, err := getKubernetesAPI(config)
 	if err != nil {
-		return err
+		return s.LogAndShareError("Scan failed while getting Kubernetes API. RunId: [%s], with reason %v", FAILED, err, kubernetesConfig.ID, workspaceId, accessToken)
 	}
+
 	log.Info("Retrieved kubernetes config Successfully")
+
 	err = s.ShareStatus(kubernetesConfig.ID, workspaceId, accessToken, IN_PROGRESS, "Retrieved Kubernetes configuration Successfully")
 	if err != nil {
-		log.Errorf("Scan failed while posting status. RunId: [%s], with reason %v", s.runId, err)
+		log.Errorf(StatusErrorFormat, s.runId, err)
+		return err
 	}
 	mapper := NewMapper(kubernetesAPI, kubernetesConfig.Cluster, workspaceId, kubernetesConfig.BlackListedNamespaces, s.runId)
 
-	var scannedObjects []DiscoveryItem
-	deployments, err := mapper.GetDeployments()
+	nodes, err := kubernetesAPI.Nodes()
 	if err != nil {
-		log.Errorf("Scan failed while fetching deployments. RunId: [%s], with reason %v", s.runId, err)
-		err = s.ShareStatus(kubernetesConfig.ID, workspaceId, accessToken, FAILED, err.Error())
-		if err != nil {
-			log.Errorf("Scan failed while posting status. RunId: [%s], with reason %v", s.runId, err)
-		}
-		return err
+		return s.LogAndShareError("Scan failed while retrieving k8s cluster nodes. RunId: [%s], with reason %v", FAILED, err, kubernetesConfig.ID, workspaceId, accessToken)
 	}
 
-	scannedObjects = append(scannedObjects, deployments...)
-	scannedObjectsByte, err := storage.Marshal(scannedObjects)
+	clusterDTO, err := mapper.GetCluster(kubernetesConfig.Cluster, nodes)
 	if err != nil {
-		log.Errorf("Scan failed while Unmarshalling results. RunId: [%s], with reason %v", s.runId, err)
-		err = s.ShareStatus(kubernetesConfig.ID, workspaceId, accessToken, FAILED, err.Error())
-		if err != nil {
-			log.Errorf("Scan failed while posting status. RunId: [%s], with reason %v", s.runId, err)
-		}
-		return err
+		return s.LogAndShareError("Scan failed while aggregating cluster information. RunId: [%s], with reason %v", FAILED, err, kubernetesConfig.ID, workspaceId, accessToken)
 	}
-	err = s.api.PostResults(scannedObjectsByte, accessToken)
+
+	// Aggregate cluster information for the event
+	namespaces, err := kubernetesAPI.Namespaces(kubernetesConfig.BlackListedNamespaces)
 	if err != nil {
-		log.Errorf("Scan failed while posting results. RunId: [%s], with reason %v", s.runId, err)
-		err = s.ShareStatus(kubernetesConfig.ID, workspaceId, accessToken, FAILED, err.Error())
-		if err != nil {
-			log.Errorf("Scan failed while posting status. RunId: [%s], with reason %v", s.runId, err)
-		}
-		return err
+		return s.LogAndShareError("Scan failed while retrieving k8s namespaces. RunId: [%s], with reason %v", FAILED, err, kubernetesConfig.ID, workspaceId, accessToken)
+	}
+
+	events, err := s.ScanNamespace(kubernetesAPI, mapper, namespaces.Items, clusterDTO, workspaceId)
```

Comment:
If I understand correctly we are in the end sending a list of events in one object. Somehow I understood from the discussion before that everything was supposed to be in one event. Is still structurally still considered as sending one event in regards to the mapping inbox?

Reply:
We changed it just now. We are sending all k8s entities for one namespace in one event now 👍